airflow/sensors/smart_sensor.py

```diff
@@ -391,13 +391,21 @@ def _update_ti_hostname(self, sensor_works, session=None):
         :param session: The sqlalchemy session.
         """
         TI = TaskInstance
-        ti_keys = [(x.dag_id, x.task_id, x.execution_date) for x in sensor_works]
 
         def update_ti_hostname_with_count(count, ti_keys):
             # Using or_ instead of in_ here to prevent from full table scan.
             tis = (
                 session.query(TI)
-                .filter(or_(tuple_(TI.dag_id, TI.task_id, TI.execution_date) == ti_key for ti_key in ti_keys))
+                .filter(
+                    or_(
+                        and_(
+                            TI.dag_id == ti_key.dag_id,
+                            TI.task_id == ti_key.task_id,
+                            TI.execution_date == ti_key.execution_date,
+                        )
+                        for ti_key in ti_keys
+                    )
+                )
```

Comment:
Can you make this stay as it was for non-mssql please? Big queries with lots of x OR y OR z is slow to build.

Reply:
done, restricted this change to mssql only